src/codecs/ico/decoder.rs

```diff
@@ -332,13 +332,15 @@ impl<'a, R: 'a + Read + Seek> ImageDecoder<'a> for IcoDecoder<R> {
                 let mask_start = r.seek(SeekFrom::Current(0))?;
                 let mask_end =
                     u64::from(self.selected_entry.image_offset + self.selected_entry.image_length);
-                let mask_length = mask_end - mask_start;
 
-                if mask_length > 0 {
+                if mask_end > mask_start {
```

Comment:
Forgive me the potentially naive question, I don't know much about the `ico` format (nor bmp really). What about the case where `mask-end == mask_start`? Is it allowed? Should it enter this block?

Reply:
No problem at all, I'm not 100% sure myself but I think `mask-end == mask-start` is what is expected if there is no mask. I've added a comment and changed the check a bit to be more precise.